Feature/2035-V2 - Folder structure refactoring

Frankaus opened the pull request:
closes #2035

This PR replaces the previous pr https://github.com/protofire/omen-exchange/pull/2144

I created various subfolder inside the market/sections/common_sections folder to better organise common components used across the market components

This PR is highly subjective so pls let me know if you find the naming and reordering right. I will edit as you request. The folder src/components/market/common has only 5 folders now, they have moved mostly inside app/src/components/market/sections/common_sections where there are 12 subfolders reoarganized as I deemed more logical.

Frankaus commented:
You can find the directory tree diagram file here https://github.com/protofire/omen-exchange/pull/2147/files#diff-ed01158c76b48af3e706f289def5cffe66abf2ab3c65ffd33659788b4bcc381d

[DIRECTORY_TREE_OMEN.pdf](https://github.com/protofire/omen-exchange/files/6924356/DIRECTORY_TREE_OMEN.pdf)

Let me know when/if you'd like me to delete the file from the repo

Frankaus commented:
We made an extra massive reordering with @Mi-Lan where we moved some files used singularly in components in the folders where they're used and deleted some unused files

https://github.com/protofire/omen-exchange/pull/2147/commits/ea0d363a7429c91f46338c946bc5c9eccb56fdd5

pimato commented:
- [ ] Categorical markets
    - [ ] Create market
    - [ ] Buy
    - [ ] Sell
    - [ ] Deposit
    - [ ] Withdraw
    - [ ] Bond outcome
    - [ ] Redeem winnings
- [ ] Scalar markets
    - [ ] Create market
    - [ ] Buy
    - [ ] Sell
    - [ ] Deposit
    - [ ] Withdraw
    - [ ] Bond outcome
    - [ ] Redeem winnings
    - [ ] Position table works as expected
        - Correct payout/profit/loss
    - [ ] Market buy slider works as expected
        - Correct payout/profit/loss
- [ ] Account
    - [ ] Correct wallet balances
    - [ ] Correct omen account balances
- [ ] xDai relay
    - [ ] Deposit
    - [ ] Withdraw
- [ ] Mobile
    - Should just be relatively functional as per usual

Mi-Lan reviewed (approved):
GJ @Frankaus its looking so much cleaner and ordered now! Make sure to delete directory _tree.md before we merge it!

Mi-Lan commented on app/src/components/market/common_sections/card_top_details/market_top_details_closed/index.tsx:
```diff
@@ -3,25 +3,44 @@ import React, { useState } from 'react'
 import { ProgressBarToggle } from '../progress_bar/toggle'
+import { SubsectionTitleWrapper } from '../subsection_title_wrapper'
+
+// import { useConnectedWeb3Context } from '../../../../contexts'
+// import { useContracts } from '../../../../hooks'
+// import { useGraphMarketsFromQuestion } from '../../../../hooks/graph/useGraphMarketsFromQuestion'
+// import { useTheme } from '../../../../hooks/useTheme'
+// import { useWindowDimensions } from '../../../../hooks/useWindowDimensions'
+// import { getContractAddress, getNativeAsset, getWrapToken } from '../../../../util/networks'
+// import {
+//   bigNumberToString,
+//   getInitialCollateral,
+//   getMarketRelatedQuestionFilter,
+//   onChangeMarketCurrency,
+// } from '../../../../util/tools'
+// import { MarketMakerData, MarketState, Token } from '../../../../util/types'
+// import { SubsectionTitleWrapper } from '../../../common'
+// import { AdditionalMarketData } from '../additional_market_data'
+// import { MarketData } from '../market_data'
+// import { ProgressBar } from '../progress_bar'
+// import { ProgressBarToggle } from '../progress_bar/toggle'
```
what is this?

KadenZipfel replied:
Looks like just got left behind from the merge... I've removed these lines

KadenZipfel commented on DIRECTORY_TREE.md:
```diff
@@ -0,0 +1,1705 @@
+│  │  │  └─ market_sections
+│  │  │     ├─ market_buy_container.tsx
+│  │  │     ├─ market_pool_liquidity_container.tsx
+│  │  │     └─ market_sell_container.tsx
```
I think we should also include the other `market_..._container` components here as well, e.g. `market_bond_container`

KadenZipfel reviewed (approved):
Hope you don't mind, I moved the rest of the market container files into the `market_sections` directory and deleted the directory tree. Should be good to go now

Frankaus commented:
> Hope you don't mind, I moved the rest of the market container files into the `market_sections` directory and deleted the directory tree. Should be good to go now

Not at all! Also thank you for deleting the comments. Glad it the reordering worked out well:)) Thanks again @Mi-Lan for the help!
